cmd/soroban-rpc/internal/daemon/daemon.go

```diff
@@ -328,32 +378,39 @@ func (d *Daemon) mustInitializeStorage(cfg *config.Config) *feewindow.FeeWindows
 			currentSeq = txMeta.LedgerSequence()
 			if initialSeq == 0 {
 				initialSeq = currentSeq
-				d.logger.WithField("seq", currentSeq).
-					Info("initializing in-memory store")
+				d.logger.
+					WithField("first", initialSeq).
+					WithField("last", ledgerSeqRange.Last).
+					Info("Initializing in-memory store")
 			} else if (currentSeq-initialSeq)%inMemoryInitializationLedgerLogPeriod == 0 {
-				d.logger.WithField("seq", currentSeq).
-					Debug("still initializing in-memory store")
+				d.logger.
+					WithField("seq", currentSeq).
+					WithField("last", ledgerSeqRange.Last).
+					Debug("Still initializing in-memory store")
 			}
 
-			if err = feeWindows.IngestFees(txMeta); err != nil {
-				d.logger.WithError(err).Fatal("could not initialize fee stats")
+			if feeStatsRange.IsLedgerIncluded(currentSeq) { // skip irrelevant ledgers
+				if err = feeWindows.IngestFees(txMeta); err != nil {
```

Comment:
you could implement the `Migration` interface for `feeWindows` and then you can expose an `Append()` function on `MultiMigration` so that the `feeWindows` migration can be included in dataMigrations.  The benefit of refactoring the code this way is that you can get rid of the fee stats special case here because everything would be encapsulated in `dataMigrations.Apply()`.  That would also allow you to get rid of:

`ledgerSeqRange := feeStatsRange.Merge(&applicableRange)`

because the multi migration applicable range would include the fee stats range

Reply:
Actually that's not true since you just need the `db.Migration` interface. Done in dce46af, is that what you had in mind?

Reply:
@Shaptic yeah, that's exactly what I had in mind, thanks!